File: Bindings~/java/src/main/java/com/alttester/Commands/UnityCommand/AltUnloadScene.java
/*
    Copyright(C) 2026 Altom Consulting
*/

package com.alttester.Commands.UnityCommand;

import com.alttester.Commands.AltBaseCommand;
import com.alttester.IMessageHandler;

public class AltUnloadScene extends AltBaseCommand {

  private AltUnloadSceneParams params;

  public AltUnloadScene(IMessageHandler messageHandler, AltUnloadSceneParams params) {
    super(messageHandler);
    this.params = params;
    params.setCommandName("unloadScene");
  }

  public void Execute() {
    SendCommand(params);
    String data = recvall(params, String.class);
    validateResponse("Ok", data);

    data = recvall(params, String.class);
    validateResponse("Scene Unloaded", data);
  }
}

File: Assets/AltTester/Runtime/AltDriver/Commands/UnityCommands/AltUnloadScene.cs
/*
    Copyright(C) 2024 Altom Consulting

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program. If not, see <https://www.gnu.org/licenses/>.
*/

namespace AltTester.AltTesterUnitySDK.Driver.Commands
{
    public class AltUnloadScene : AltBaseCommand
    {
        AltUnloadSceneParams cmdParams;
        public AltUnloadScene(IDriverCommunication commHandler, string sceneName) : base(commHandler)
        {
            cmdParams = new AltUnloadSceneParams(sceneName);
        }
        public void Execute()
        {
            CommHandler.Send(cmdParams);

            var data = CommHandler.Recvall<string>(cmdParams);
            ValidateResponse("Ok", data);

            data = CommHandler.Recvall<string>(cmdParams);
            ValidateResponse("Scene Unloaded", data);
        }
    }
}
